test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs

```diff
@@ -1073,10 +1095,11 @@ public Task ResponseStatusCodeSetBeforeHttpContextDisposedRequestMalformedReadIg
                 {
                     await connection.Send(
                         "POST / HTTP/1.1",
+                        "Host:",
                         "Transfer-Encoding: chunked",
                         "",
                         "gg");
-                    await responseWritten.WaitAsync();
+                    await responseWritten.WaitAsync().TimeoutAfter(TimeSpan.FromSeconds(10));
```

Comment:
I'd use 30s then. It only require more patience if it fails, and potentially makes the test less flaky.